crates/store/re_video/src/decode/ffmpeg.rs

```diff
@@ -0,0 +1,730 @@
+
+    #[error("Bad video data: {0}")]
+    BadVideoData(String),
+
+    #[error("FFMPEG error: {0}")]
+    Ffmpeg(String),
+
+    #[error("FFMPEG fatal error: {0}")]
+    FfmpegFatal(String),
+
+    #[error("FFMPEG IPC error: {0}")]
+    FfmpegSidecar(String),
+
+    #[error("FFMPEG exited unexpectedly with code {0:?}")]
+    FfmpegUnexpectedExit(Option<std::process::ExitStatus>),
+
+    #[error("Failed to send video frame info to the ffmpeg read thread.")]
+    BrokenFrameInfoChannel,
+}
+
+impl From<Error> for super::Error {
+    fn from(err: Error) -> Self {
+        Self::Ffmpeg(std::sync::Arc::new(err))
+    }
+}
+
+/// ffmpeg does not tell us the timestamp/duration of a given frame, so we need to remember it.
+#[derive(Clone)]
+struct FfmpegFrameInfo {
+    presentation_timestamp: Time,
+    duration: Time,
+    decode_timestamp: Time,
+}
+
+enum FfmpegFrameData {
+    Chunk(Chunk),
+    EndOfStream,
+}
+
+struct FfmpegProcessAndListener {
+    ffmpeg: FfmpegChild,
+
+    /// For sending frame timestamps to the ffmpeg listener thread.
+    frame_info_tx: Sender<FfmpegFrameInfo>,
+
+    /// For sending chunks to the ffmpeg write thread.
+    frame_data_tx: Sender<FfmpegFrameData>,
+
+    listen_thread: Option<std::thread::JoinHandle<()>>,
+    write_thread: Option<std::thread::JoinHandle<()>>,
+}
+
+impl FfmpegProcessAndListener {
+    fn new(
+        debug_name: &str,
+        on_output: Arc<OutputCallback>,
+        avcc: re_mp4::Avc1Box,
+    ) -> Result<Self, Error> {
+        re_tracing::profile_function!();
+
+        let mut ffmpeg = FfmpegCommand::new()
+            // "Reduce the latency introduced by buffering during initial input streams analysis."
+            //.arg("-fflags nobuffer")
+            //
+            // .. instead use these more aggressive options found here
+            // https://stackoverflow.com/a/49273163
+            .args([
+                "-probesize",
+                "32", // 32 bytes is the minimum probe size.
+                "-analyzeduration",
+                "0",
+            ])
+            // Keep in mind that all arguments that are about the input, need to go before!
+            .format("h264") // TODO(andreas): should we check ahead of time whether this is available?
+            //.fps_mode("0")
+            .input("-") // stdin is our input!
+            // h264 bitstreams doesn't have timestamp information. Whatever ffmpeg tries to make up about timing & framerates is wrong!
+            // If we don't tell it to just pass the frames through, variable framerate (VFR) video will just not play at all.
+            .fps_mode("passthrough")
+            // TODO(andreas): at least do `rgba`. But we could also do `yuv420p` for instance if that's what the video is specifying
+            // (should be faster overall at no quality loss if the video is in this format).
+            // Check `ffmpeg -pix_fmts` for full list.
+            .rawvideo() // Output rgb24 on stdout.
+            .spawn()
```

Comment:
My comment on process starts being slow was entirely about the side of the new process :). But I'll make sure to give this another spin on Windows and check on this!
Moving it to background thread is a but cumbersome and of potentially dubious value because we need the process object & handle on the main thread potentially immediately